worlds/stardew_valley/logic/grind_logic.py

```diff
@@ -31,26 +33,28 @@ def __init__(self, *args, **kwargs):
         self.grind = GrindLogic(*args, **kwargs)
 
 
-class GrindLogic(BaseLogic[Union[GrindLogicMixin, HasLogicMixin, ReceivedLogicMixin, BookLogicMixin, TimeLogicMixin, ToolLogicMixin]]):
+class GrindLogic(BaseLogic[Union[GrindLogicMixin, HasLogicMixin, ReceivedLogicMixin, RegionLogicMixin, BookLogicMixin, TimeLogicMixin, ToolLogicMixin]]):
 
     def can_grind_mystery_boxes(self, quantity: int) -> StardewRule:
+        opening_rule = self.logic.region.can_reach(Region.blacksmith)
         mystery_box_rule = self.logic.has(Consumable.mystery_box)
         book_of_mysteries_rule = self.logic.true_ \
             if self.options.booksanity == Booksanity.option_none \
             else self.logic.book.has_book_power(Book.book_of_mysteries)
         # Assuming one box per day, but halved because we don't know how many months have passed before Mr. Qi's Plane Ride.
         time_rule = self.logic.time.has_lived_months(quantity // 14)
-        return self.logic.and_(mystery_box_rule,
-                               book_of_mysteries_rule,
-                               time_rule)
+        return self.logic.and_(opening_rule, mystery_box_rule,
+                               book_of_mysteries_rule, time_rule,)
```

Comment:
Any reason this isn't just one line?

Reply:
Not particularly, any reason it should be?

Reply:
Just seems odd to me, since it's a different formatting from before and also it has the final `,` which none of the other lines in this file do

Reply:
The previous formatting was one argument per line to minimize potential git conflicts. We should stick to it or change it to the standard "all arguments on the same line". Two arguments is quite unconventional...